src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/GeneratedCodeContainer.cs

```diff
@@ -161,31 +165,30 @@ public void SetOutput(RazorCSharpDocument csharpDocument, DefaultDocumentSnapsho
         // Internal for testing.
         internal static bool TryGetLinePositionSpan(TextSpan span, SourceText source, RazorCSharpDocument output, out LinePositionSpan linePositionSpan)
         {
-            for (var i = 0; i < output.SourceMappings.Count; i++)
+            var mappings = output.SourceMappings;
+            for (var i = 0; i < mappings.Count; i++)
             {
-                var mapping = output.SourceMappings[i];
-                if (span.Length > mapping.GeneratedSpan.Length)
-                {
-                    // If the length of the generated span is smaller they can't match. A C# expression
-                    // won't cover multiple generated spans.
-                    //
-                    // This heuristic is useful in the Razor context to filter out zero-length
-                    // spans.
-                    continue;
-                }
```

Comment:
This logic is wrong. It's only comparing the sizes of the spans, and isn't looking at the positions. If only there was a method for that....